cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -356,6 +361,20 @@ protected static ProgramRecord makeProgramRecord(String appId, ProgramSpecificat
     return null;
   }
 
+  protected Map<RunId, ProgramRuntimeService.RuntimeInfo> findAllRuntimeInfos(String namespaceId, String appId,
```

Comment:
Added new method in the `ProgramRuntimeService` to `list` the runtime info objects for given `Id.Program`